Java-Inheritance.md

```diff
@@ -0,0 +1,168 @@
+
+## Override it!
+
+Java lets you override or redefine the methods defined in the superclass. For example, your `Car` class has a different implementation of `start()` than the parent `Vehicle`, so you do this :
+
+```java
+public class Vehicle() {
+	public void start() {
+	  System.out.println("Vehicle start code");
+	}
+}
+
+public class Car extends Vehicle {
+	public void start() {
+	  System.out.println("Car start code");
+  }
+}
+
+Car car = new Car();
+car.start(); // "Car start code"
+```
+
+So, its pretty simple to override methods in the subclass. Although, there is a _catch_. Only that superclass method will be overriden which has the _exact same method signature_ as the subclass method. Thus, `public void start(String key)` would not override `public void start()`.
+
+**Notes** :
+
+* You cannot override private methods of the superclass. (Quite obvious, isn't it?)
+* What if the method of superclass which you are overriding in the subclass suddenly gets obliterated or methods changed? It would fail in runtime! So Java provides you a nifty annotation `@Override` which you can place over the subclass method, which will warn the compiler of those incidents!
+
+## How to call super class methods?
+
+Funny you ask about it! Just use the keyword `super` :
+
+```java
+public class Vehicle() {
+	public void start() {
+	  System.out.println("Vehicle start code");
+	}
+}
+
+public class Car extends Vehicle {
+	public void run() {
+	  super.start();
+  }
+}
+
+Car car = new Car();
+car.run(); // "Vehicle start code"
+```
+
+## How to know the type of a class?
+
+Using the `instanceof` keyword. Having lots of classes and subclasses it would be a little confusing to know which class is a subclass of which one in runtime. So, we can use `instanceof` to determine whether a class is actually a subclass of a superclass or not.
+
+```java
+Car car = new Car();
+
+boolean flag = car instanceof Vehicle; // true in this case!
+```
+
+## Constructors & Inheritance
+
+As mentioned earlier, constructors cannot be directly inherited by a subclass. Although, a subclass is _required_ to call its parent's constructor as the first thing in its own constructor. How? You guessed it, using `super` :
+
+```java
+public class Vehicle() {
+	public Vehicle() {
+		// constructor
+	}
+	public void start() {
+	  System.out.println("Vehicle start code");
+	}
+}
+
+public class Car extends Vehicle {
+	public Car() {
+		super();
+	}
+	public void run() {
+	  super.start();
+  }
+}
+```
+
+Remember, if the superclass does not have any constructors defined, you dont have to call it explicitely in the subclass. Java handles that internally for you!
+
+Congrats, now you know all about Inheritance! Read more about advanced ways to inherit things in [Abstract Classes]() and [Interfaces]()!
```

Comment:
@abhisekp take a look at this line. Should these links be blank, because the articles are not yet added, but would be added soon.